plugins/org.jboss.tools.openshift.ui/src/org/jboss/tools/openshift/internal/ui/server/OpenShiftServerEditorSection.java

```diff
@@ -452,302 +240,403 @@ private void createOpenShiftApplicationControls(FormToolkit toolkit, Composite p
+						deployPathText.getText(), OpenShiftServerUtils.ATTR_POD_PATH, this));
+			}
+		}; 
+		ModifyListener deployPathDecoration = new TextboxDecoratorSupport(deployPathText, new IValidator() {
+			@Override
+			public IStatus validate(Object value) {
+				// TODO validate 
+				if (!(value instanceof String) || StringUtils.isEmpty(value)) {
+					return ValidationStatus.cancel("Please provide a path to deploy to on the pod.");
+				}
+				if (!Path.isValidPosixPath((String)value)) {
+					return ValidationStatus.error("Please provide a valid path to deploy to on the pod");
 				}
-				
-			})
-			.to(BeanProperties.value(OpenShiftServerEditorModel.PROPERTY_POD_PATH).observe(model))
-			.in(dbc);
+				return Status.OK_STATUS;
+			}
+		});
+		deployPathText.addModifyListener(deployPathDecoration);
 	}
 
 	@SuppressWarnings("unchecked")
 	private void createResourceControls(Composite parent, DataBindingContext dbc) {
-		Label kindLabel = new Label(parent, SWT.None);
-		GridDataFactory.fillDefaults()
-			.align(SWT.FILL, SWT.CENTER)
-			.applyTo(kindLabel);
-		ValueBindingBuilder
-			.bind(WidgetProperties.text().observe(kindLabel))
-			.notUpdatingParticipant()
-			.to(BeanProperties.value(OpenShiftServerEditorModel.PROPERTY_RESOURCE).observe(model))
-			.converting(new Converter(IResource.class, String.class) {
-
-				@Override
-				public Object convert(Object fromObject) {
-					if (!(fromObject instanceof IResource)) {
-						return "Resource:";
-					}
-					return ((IResource) fromObject).getKind() + ":";
-				}
-			})
-			.in(dbc);
-
-		Text resourceText = new Text(parent, SWT.BORDER | SWT.READ_ONLY);
-        GridDataFactory.fillDefaults()
-			.align(SWT.FILL, SWT.CENTER)
-			.grab(true, false)
-			.applyTo(resourceText);
-		ValueBindingBuilder
-			.bind(WidgetProperties.text(SWT.Modify).observe(resourceText))
-			.to(BeanProperties.value(OpenShiftServerEditorModel.PROPERTY_RESOURCE).observe(model))
-			.converting(new Converter(IResource.class, String.class) {
-				
-				@Override
-				public Object convert(Object fromObject) {
-					if (!(fromObject instanceof IResource)) {
-						return "<not found>";
-					};
-					IResource resource = (IResource) fromObject;
-					return resource.getNamespace() + "/" + resource.getName();
-				}
-			})
-			.in(dbc);
-
-		Button selectResourceButton = new Button(parent, SWT.PUSH);
-		selectResourceButton.setText("Select...");
-		GridDataFactory.fillDefaults()
-				.align(SWT.FILL, SWT.CENTER).hint(120, SWT.DEFAULT)
-				.applyTo(selectResourceButton);
-		selectResourceButton.addSelectionListener(onSelectResource(model, selectResourceButton.getShell()));
+		resourceKindLabel = new Label(parent, SWT.None);
+		applyAlignData(SWT.FILL, SWT.CENTER, resourceKindLabel);
+		resourceText = new Text(parent, SWT.BORDER | SWT.READ_ONLY);
+		applyGridData(2, 1, SWT.FILL, SWT.CENTER, true, false, resourceText);
```

Comment:
that's really ugly. There's no way for you to know what these parameters are about without getting to the method and reading what it really does.